content-repo/gen_top_contrib.py

```diff
@@ -293,10 +354,31 @@ def main():
     args = parser.parse_args()
     contrib_target = args.target + '/top-contributors.md'
     try:
-        response = get_inner_pr_request()
-        users_list = get_contributors_users(response)
+        service_account_file = tempfile.NamedTemporaryFile(delete=False, mode='w', suffix='.json')
+        json.dump(json.loads(SERVICE_ACCOUNT), service_account_file)
+        service_account_file.flush()
+        contributors_data = get_contributors_file_from_bucket(service_account_file.name)
+        query = 'type:pr state:closed org:demisto repo:content is:merged base:master head:contrib/ sort:updated-desc'
+        # First time upload or in a case of malfunction
+        if contributors_data:
+            # Update the file with recent contributions
+            contributors_data = json.loads(contributors_data)
+            last_updated_timestamp = contributors_data.get('last_update')
+            query += f' merged:>{last_updated_timestamp}'
+
+        response, last_updated = get_inner_pr_request(query)
+        users_dict = get_contributors_users(response, last_updated)
+
+        if contributors_data:
+            users_dict = update_users_list(users_dict, contributors_data)
+
+        print('Updating contributors file...')
+        update_contributors_file(service_account_file.name, json.dumps(users_dict, indent=4))
+        users_dict.pop('last_update')
+        sorted_users_list = sorted(users_dict.items(), key=lambda x: x[1]['number_of_contributions'], reverse=True)
```

Comment:
Let's move this out from main into the create users list function

Reply:
I prefer not to change the flow here, the `create_users_list` function expects a list.
Changing this will mean rewriting the function which I don't think is necessary.

Reply:
Not sure I follow, what is the problem that `create_users_list` will receive a dict and will return a list? 
Your changes make the main function much less readable, which is why we use functions for abstraction.